src/jitlayers.cpp

```diff
@@ -422,3 +449,555 @@ class JuliaOJIT {
 
 }
 #endif
+
+#ifdef USE_ORCJIT
+JuliaOJIT *jl_ExecutionEngine;
+#else
+ExecutionEngine *jl_ExecutionEngine;
+#endif
+
+// destructively move the contents of src into dest
+// this assumes that the targets of the two modules are the same
+// including the DataLayout and ModuleFlags (for example)
+// and that there is no module-level assembly
+static void jl_merge_module(Module *dest, std::unique_ptr<Module> src)
```

Comment:
breadcumb: in the future, we may want to just call `Linker::linkModules` (introduced in llvm-3.8 for LTO), but for now, our actual usage is much easier than that because we don't need full generality (or barely any at all), so we can just do the simple thing here